Expensify/App
# [No QA] Only load full policies one-by-one as needed

### Details
So we want to make sure that whenever a workspace or any of its sub-pages attempts to render, then we'll load the full policy for that workspace. So whenever we navigate to a workspace page (and did not just come from one), then we'll load that full policy.

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/181676

### Tests
1. Sign into NewDot using an account with a workspace.
1. Go to `/settings`. Verify using logs or the network console that no full policy is loaded.
1. Click on an existing workspace. Verify using logs or the network console that the full policy is loaded.
1. Refresh the page. Verify using logs or the network console that the full policy is loaded.
1. Click on the back arrow. Verify you are taken back to the settings page.
1. Click on an existing workspace. Verify using logs or the network console that the full policy is loaded.
1. Click on a workspace sub-page. Verify using logs or the network console that no full policy is loaded.
1. Refresh the page on one of the workspace sub-pages. Verify using logs or the network console that the full policy is loaded.
1. Click on the `x` to close the modal. Verify that the full settings modal closes.


### QA Steps
Only regression tests.

### Tested On
